File: Day-5/KBalls.js
function canArrange(arr, k) {
  if (arr.length % k !== 0) return false;

  const count = {};
  for (let num of arr) {
      count[num] = (count[num] || 0) + 1;
  }

  const keys = Object.keys(count).map(Number).sort((a, b) => a - b);

  for (let num of keys) {
      let freq = count[num];
      if (freq > 0) {
          for (let i = 0; i < k; i++) {
              let current = num + i;
              if (!count[current] || count[current] < freq) {
                  return false;
              }
              count[current] -= freq;
          }
      }
  }

  return true;
}

// Test cases here

File: Day-5/KBalls.cpp
#include <iostream>
#include <vector>
#include <map>
#include <algorithm>

bool canArrange(std::vector<int>& arr, int k) {
    if (arr.size() % k != 0) return false;

    std::map<int, int> count;
    for (int num : arr) {
        count[num]++;
    }

    for (auto& [num, freq] : count) {
        if (freq > 0) {
            for (int i = 0; i < k; i++) {
                int current = num + i;
                if (count[current] < freq) {
                    return false;
                }
                count[current] -= freq;
            }
        }
    }

    return true;
}

int main() {
    // Test cases here
    return 0;
}
